src/app/shared/components/template/components/layout/nav_group.ts

```diff
@@ -100,5 +106,36 @@ export class NavGroupComponent extends TemplateLayoutComponent {
 
   goToSection(index: number) {
     this.sectionIndex = index;
+    this.updateSectionProgress();
+  }
+
+  updateSectionProgress() {
+    //update the field provided in progress_variable to be equal to the max of it's current value
+    //and the percentage of this.sectionIndex from this.templateNames.length. the value should
+    //be an integer between 0 and 100 inclusive.
+    const progressField = this._row.parameter_list["progress_field"];
+    if (progressField && progressField.indexOf("{{") < 0) {
+      const currentPercentDone = Math.ceil(
+        ((this.sectionIndex + 1) / this.templateNames.length) * 100
+      );
+      const previousPercentDone = Number.parseInt(this.templateService.getField(progressField));
```

Comment:
You can probably drop the need for extra checks if you specify a fallback for the getField, i.e:
`parseInt(this.templateService.getField(progressField) || 0)`

Also I don't think you need to use both the number constructor and parseInt, unless there's some weird edge case I'm not aware of